src/server/req_rescq.c

```diff
@@ -773,11 +776,15 @@ req_confirmresv(struct batch_request *preq)
 	}
 	rc = assign_resv_resc(presv, next_execvnode, FALSE);
 
-	if (presv->ri_qs.ri_stime < time_now) {
+	DBPRT(("resv_name=%s, rc=%d, is_degraded=%d, stime=%ld, now=%ld",
+	       presv->ri_qs.ri_resvID, is_degraded, rc, presv->ri_qs.ri_stime, time_now));
+
+	if (presv->ri_qs.ri_stime <= time_now) {
```

Comment:
I vaguely remember having an issue if a reservation is reconfirmed RIGHT as the reservation is started.  I couldn't figure out why, and was pressed for time, so I punted.  I just made it so the server wouldn't tell the scheduler to reconfirm if the reservation just started.   Could this be the issue?

Reply:
That was one of the issues, which is what was causing test_resv_server_restart to ocassionally fail.  I'm not how you avoided reconfirmation at start time since my understanding is that a recently degraded reservation needs to be reconfirmed before being started.

The other issue was when the reservation was degraded, and new resources were assigned, the state of those resources wasn't being updated, leaving them in the free state.  test_resv_exclusive_after_node_offlined was added in this PR to consistently demonstrate the state issue.